### ISSUE TITLE
[Feature suggestion] Let ApkTrack take the Android version into account

### ISSUE THREAD

**Issue by Olf0** [JusticeRage/ApkTrack#109]
Please let ApkTrack take the Android version into account, when determining which Apps are upgradable.
ApkTrack 2.1.1 displays a lot of upgradable apps under older Android versions, but when trying to upgrade, that often fails due to the apk being built for newer Android API levels.
Hence it is a quite tedious work to find the really upgradable apps (which is ApkTrack's primary task, IMO ;) ) on older Android versions. 

Tested with ApkTrack 2.1.1 under AOSP 4.1.2

Please let me know, if logs etc. might be helpful.

**Comment by JusticeRage** [comment]
That would require major alterations to the way ApkTrack currently works. I'm currently considering dropping support for the app altogether, considering an alternative exists (APKUpdater I believe) and is more actively maintained / feature rich.
I have to give more thought to whether ApkTrack is still needed before starting to work on that.

**Comment by Olf0** [comment]
O.K. 
I expected this to be an intrusive change and thus originally hesitated to file a feature request for it.  Well, ultimately thought it would be worth trying, at least. 

Thanks for pointing to APKupdater as an alternative, I will take a look at it. 
Side note: Ugh, yesterday I was so happy to have found this excellent app (despite its minor flaws), now you make me consider using a different app. ;) 
My first thought was, "Oh, please don't, ApkTrack is extremely useful!".  Well, I'll see, which feature set APKupdater exactly provides. 

P.S.: It took me a while to find APKUpdater (com.apkupdater) / "Apk Updater Apk installer" by rumboalla, [releases](https://www.apkmirror.com/apk/rumboalla/apkupdater/apkupdater-1-5-11-release/apkupdater-1-5-11-android-apk-download/download/) and [source code repository](https://github.com/rumboalla/apkupdater). 

P.P.S.: APKUpdater has a couple of very nice features, but also many features lacking, which ApkTrack provides (e.g. access to f-droid etc.).  Furthermore accessing APKpure and uptodown is broken in APKUpdater 1.5.11: It crashes, when looking for upgradeable apps there. 
Hmm, I am not yet convinced, that APKUpdater is (yet) a suitable alternative to ApkTrack.

**Comment by JusticeRage** [comment]
Thanks for your feedback. As I don't use APKUpdater, I'm not sure yet how it compares to ApkTrack.

**Comment by Olf0** [comment]
> As I don't use APKUpdater,

That's why I provided above web-links to APKUpdater's releases (and sources) for you and others. :)

> I'm not sure yet how it compares to ApkTrack. 

Well, ApkTrack and APKUpdater are surprisingly different, for both basically pursuing the same goal:

- They use completely different APK sources, except for the PlayStore: 
   a. ApkTrack is much more oriented towards Free Software (which I like), with F-droid as its best working and integrated APK source, PlayStore, ApkTrack Proxy for APK sources with (also) proprietary software, Xposed stable, Xposed beta, plus (when detected) the Guardian Project, Öffi and maybe others. 
   b. APKUpdater uses APKmirror as its best working and integrated APK source, PlayStore, and (optionally, currently not working) APKpure, Uptodown. 

- ApkTrack's list view of APKs has a much better UI (i.e. "Look & Feel"):
   a. One entry per APK with selectable sources, instead of (APKUpdater) one entry per hit in any of the selected sources, resulting in multiple (consecutive) entries for one installed APK. 
   b. Installed APKs can be individually (or all) queried for available updates (ApkTrack), in contrast to (APKUpdater) always all installed APKs must be queried. 
   c. All installed APKs are displayed in selectable order (ApkTrack) vs. only updateable APKs are displayed (APKUpdater).

- OTOH, APKUpdater promises to take the API level of the installed Android into account (at least when querying APKmirror), but that does not seem to work properly (yet). 

- APKUpdater also displays the original version strings (along a human readable version number), which is very helpful; a feature ApkTrack currently (as of v2.1.2) lacks. 

All in all: A combination of both apps (with a superset of each of their current features) would result in a perfect software for this purpose. ;)

**Comment by Olf0** [comment]
> That would require major alterations to the way ApkTrack currently works.

As ApkTrack is quite useful with its current design (using "web page scraping"), closing this issue (and hence recalling this suggestion), as it does not seem to be worth the effort of investigating this further.

**Comment by Olf0** [comment]
This issue is a duplicate of issue #58.
